zcash_keys/src/address.rs

```diff
@@ -342,7 +342,14 @@ impl Address {
     }
 }
 
-#[cfg(any(test, feature = "test-dependencies"))]
+#[cfg(all(
+    any(
+        feature = "orchard",
+        feature = "sapling",
+        feature = "transparent-inputs"
+    ),
+    any(test, feature = "test-dependencies")
+))]
```

Comment:
I don't think we support enabling neither `sapling` nor `orchard`, do we?

Reply:
These are feature flags, so we don't have a way (that I know of) to enforce that at least one be enabled at the build system level.

Reply:
I suspect a `#[cfg(not(any(` type attribute _might_ allow you to introduce a compile break if you were interested in that, but I haven't tried it.

Reply:
In any case, with ZIP 316 Revision 1, transparent-only UFVKs will be allowed (for use with metadata.)

Reply:
They will, but there's still no point in *none* of "orchard", "sapling", or "transparent-inputs" being enabled. That should be a compilation error IMHO.

Reply:
You could still use this crate to parse `TransparentAddress` values with metadata, so that you can send to them properly. `transparent-inputs` only applies to key and address _generation_, not parsing.

Reply:
The `transparent-inputs` flag is not any different to the`sapling` and `orchard` flags. All three have the purpose of removing the need for certain dependencies; in the case of `transparent-inputs` it is the `secp256k1` crate. It's only because we solely support P2PKH and P2SH outputs that we only flag off transparent inputs; if we had to support arbitrary transparent output generation then we'd likely need `secp256k1` for that, and we'd need a; more general transparent flag.